lib/srv/db/postgres/delete-user.sql

```diff
@@ -0,0 +1,20 @@
+create or replace procedure teleport_delete_user(username varchar)
+language plpgsql
+as $$
+declare
+    role_ varchar;
+begin
+    -- Only drop if the user doesn't have other active sessions.
+    if exists (select usename from pg_stat_activity where usename = username) then
+        raise notice 'User has active connections';
+    else
+        begin
+            execute format('drop user %I', username);
+        exception
+            when SQLSTATE '2BP01' then
+                -- Drop user/role will fail if user has dependent objects.
+                -- In this scenario, fallback into disabling the user.
+                call teleport_deactivate_user(username);
```

Comment:
curious if we are able to tell if user is dropped successfully or the user is deactivated based on our logs?

Reply:
Initially, I've added a specific return for delete and deactivate. I've removed it since this procedure is simple and only has a fallback when a particular error happens. I can bring it back if you feel this would have a good debugging value.

Reply:
> I can bring it back if you feel this would have a good debugging value.

That would be great. I think it will be valuable to tell if `preferDrop` actually drops or not.